hr_time/public/js/easy_checkin_dialog.js

```diff
@@ -1,129 +1,355 @@
+        const employee_id = response.message;
+
+        if (!employee_id) {
+          frappe.throw(__("No employee ID found for the current user"));
+          return;
         }
+
+        this.checkIfEmployeeHasWorklogs(employee_id);
+      },
+    });
+  }
+
+  /**
+   * Checks if the employee has worklogs for today using the provided employee ID.
+   * @param {string} employee_id - The ID of the employee to check for worklogs.
+   */
+  checkIfEmployeeHasWorklogs(employee_id) {
+    frappe.call({
+      method: "hr_time.api.worklog.api.has_employee_made_worklogs_today",
+      args: { employee_id: employee_id },
+      callback: (response) => {
+        this.hasWorklogs = response.message;
+        this.createCheckinDialog(employee_id);
+      },
+    });
+  }
+
+  /**
+   * Creates and displays the check-in dialog with options and actions.
+   * @param {string} employee_id - The ID of the employee.
+   */
+  createCheckinDialog(employee_id) {
+    this.dialogUI = new frappe.ui.Dialog({
+      title: `&#9745;&nbsp;${__("Checkin")}`, //checkbox-icon; space; label
+      fields: this.getDialogFields(employee_id),
+      size: "small",
+      primary_action_label: __("Submit", undefined, "checkin"),
+      primary_action: (values) => this.submitCheckin(values, employee_id),
+    });
+
+    this.dialogUI.show();
+    this.initializeDialog();
+  }
+
+  /**
+   * Returns the configuration of fields to be displayed in the check-in dialog.
+   * @param {string} employee_id - The ID of the employee.
+   * @returns {Array<Object>} - The fields configuration for Frappe's UI dialog.
+   */
+  getDialogFields(employee_id) {
+    return [
+      {
+        label: "Action",
+        fieldname: "action",
+        fieldtype: "Select",
+        options: this.options,
+        default: this.default,
+        change: () => this.updateDialogBasedOnAction(),
+      },
+      {
+        fieldtype: "Section Break",
+        depends_on: 'eval: doc.action === "End of work"',
+      },
+      {
+        fieldname: "worklog_section",
+        fieldtype: "HTML",
+        options: this.getWorklogSectionHTML(),
+      },
+      {
+        fieldname: "worklog_box",
+        fieldtype: "Text",
+        placeholder: __("Describe your task here") + "...",
+      },
+      {
+        fieldtype: "Button",
+        label: "✔",
+        fieldname: "add_log",
+        click: () => this.addWorklog(employee_id),
+      },
+    ];
+  }
+
+  /**
+   * Generates the HTML content for the worklog section of the dialog.
+   * @returns {string} - The HTML content for the worklog section.
+   */
+  getWorklogSectionHTML() {
+    return `<div class="d-flex justify-between">
+                <label id="worklog_section_label">${__("Add Worklog")}</label>
+                <button class="btn btn-outline-info btn-xs edit-full-form-btn">${__(
+                  "Enter complete detail"
+                )}
+                    &nearr;
+                </button>
+            </div>
+            `;
```

Comment:
Check if you might could use the template engine using `frappe.render_template()`.